### PR TITLE
chore(generator): IJSHandle and IElement

### LINKED ISSUE #1240: Use generated IJSHandle

### LINKED ISSUE #1241: Use generated IElementHandle

### PR DESCRIPTION
**Pull request by kblok**
closes #1240
closes #1241

### CODE REVIEW

File: src/PlaywrightSharp/IFrame.cs [hunk 1]
```diff
@@ -433,7 +433,7 @@ public interface IFrame
         /// <param name="noWaitAfter">Actions that initiate navigations are waiting for these navigations to happen and for pages to start loading.
         /// You can opt out of waiting via setting this flag. You would only need this option in the exceptional cases such as navigating to inaccessible pages. Defaults to false.</param>
         /// <returns>A <see cref="Task"/> that completes when the element matching <paramref name="selector"/> is successfully double clicked.</returns>
-        Task DblClickAsync(
+        Task DblclickAsync(
```
Comment (avodovnik) [inline src/PlaywrightSharp/IFrame.cs:436]: This is a generator problem. It should be `DblClickAsync`, so we should definitely have an upstream PR that adds a `alias-csharp: DblClickAsync` into the file.

File: src/PlaywrightSharp.Tests/ElementHandleClickTests.cs [hunk 2]
```diff
@@ -112,7 +112,7 @@ public async Task ShouldDoubleClickTheButton()
             }");
 
             var button = await Page.QuerySelectorAsync("button");
-            await button.DblClickAsync();
+            await button.DblclickAsync();
```
Comment (avodovnik) [inline src/PlaywrightSharp.Tests/ElementHandleClickTests.cs:115]: This needs to be fixed in upstream.

File: src/PlaywrightSharp.Tests/PageSetInputFilesTests.cs [hunk 3]
```diff
@@ -55,11 +55,11 @@ public async Task ShouldSetFromMemory()
         {
             await Page.SetContentAsync("<input type=file>");
 
-            await Page.SetInputFilesAsync("input", new FilePayload
+            await Page.SetInputFilesAsync("input", new ElementHandleFiles
             {
                 Name = "test.txt",
                 MimeType = "text/plain",
-                Buffer = Convert.ToBase64String(Encoding.UTF8.GetBytes("this is a test"))
+                Buffer = Encoding.UTF8.GetBytes("this is a test"),
```
Comment (avodovnik) [inline src/PlaywrightSharp.Tests/PageSetInputFilesTests.cs:62]: This is a significant change, did you add it to the draft release/breaking changes yet?

File: src/PlaywrightSharp/Contracts/IMouse.generated.cs [hunk 4]
```diff
@@ -1,4 +1,4 @@
-/*
+﻿/*
```
Comment (avodovnik) [inline src/PlaywrightSharp/Contracts/IMouse.generated.cs:1]: o.O

File: src/PlaywrightSharp/Mouse.cs [hunk 5]
```diff
@@ -13,25 +14,18 @@ public Mouse(PageChannel channel)
         }
 
         public Task ClickAsync(float x, float y, MouseButton button, int? clickCount, float? delay)
-            => _channel.MouseClickAsync(x, y, delay ?? 0, EnsureDefaultValue(button), clickCount ?? 1);
+            => _channel.MouseClickAsync(x, y, delay ?? 0, button.EnsureDefaultValue(MouseButton.Left), clickCount ?? 1);
 
         public Task DblclickAsync(float x, float y, MouseButton button, float? delay)
```
Comment (avodovnik) [inline src/PlaywrightSharp/Mouse.cs:19]: `DblClick` -> upstream?

**Review by avodovnik — CHANGES_REQUESTED** [review]

File: src/PlaywrightSharp.Tests/PageSetInputFilesTests.cs (hunk 3 above)
Comment (kblok) [reply src/PlaywrightSharp.Tests/PageSetInputFilesTests.cs:62]: Yes Sir.

File: src/PlaywrightSharp/Contracts/IMouse.generated.cs (hunk 4 above)
Comment (kblok) [reply src/PlaywrightSharp/Contracts/IMouse.generated.cs:1]: dotnet format

File: src/PlaywrightSharp/FileChooserEventArgs.cs [hunk 6]
```diff
@@ -48,7 +49,7 @@ public FileChooserEventArgs(IPage page, IElementHandle element, bool multiple)
         /// <param name="noWaitAfter">Actions that initiate navigations are waiting for these navigations to happen and for pages to start loading.</param>
         /// <returns>A <see cref="Task"/> that completes when the files are successfully set.</returns>
         public Task SetFilesAsync(string file, int? timeout = null, bool? noWaitAfter = null)
-            => Element.SetInputFilesAsync(file, timeout, noWaitAfter);
+            => Element.SetInputFilesAsync(file, noWaitAfter, timeout);
```
[the remark this answers is not in the document]
Comment (kblok) [reply src/PlaywrightSharp/FileChooserEventArgs.cs:52]: done

File: src/PlaywrightSharp/JSHandle.cs [hunk 7]
```diff
@@ -77,7 +59,7 @@ public async Task<T> EvaluateAsync<T>(string expression, object arg)
         public async Task<IJSHandle> GetPropertyAsync(string propertyName) => (await _channel.GetPropertyAsync(propertyName).ConfigureAwait(false))?.Object;
 
         /// <inheritdoc />
-        public async Task<IDictionary<string, IJSHandle>> GetPropertiesAsync()
+        public async Task<Dictionary<string, IJSHandle>> GetPropertiesAsync()
```
[the remark this answers is not in the document]
Comment (kblok) [reply src/PlaywrightSharp/JSHandle.cs:62]: I'm changing the class to return what the generated code says we should return.

File: src/PlaywrightSharp/IFrame.cs (hunk 1 above)
Comment (kblok) [reply src/PlaywrightSharp/IFrame.cs:436]: https://github.com/microsoft/playwright/pull/5899

File: src/PlaywrightSharp.Tests/ElementHandleClickTests.cs (hunk 2 above)
Comment (kblok) [reply src/PlaywrightSharp.Tests/ElementHandleClickTests.cs:115]: https://github.com/microsoft/playwright/pull/5899

File: src/PlaywrightSharp/IPage.cs [hunk 8]
```diff
@@ -1283,7 +1283,7 @@ Task<T> WaitForEventAsync<T>(PlaywrightEvent<T> pageEvent, Func<T, bool> predica
         /// <param name="noWaitAfter">Actions that initiate navigations are waiting for these navigations to happen and for pages to start loading.
         /// You can opt out of waiting via setting this flag. You would only need this option in the exceptional cases such as navigating to inaccessible pages. Defaults to false.</param>
         /// <returns>A <see cref="Task"/> that completes when the element matching <paramref name="selector"/> is successfully double clicked.</returns>
-        Task DblClickAsync(
+        Task DblclickAsync(
```
[the remark this answers is not in the document]
Comment (kblok) [reply src/PlaywrightSharp/IPage.cs:1286]: https://github.com/microsoft/playwright/pull/5899

**Review by avodovnik — APPROVED** [review]

**Review by avodovnik — APPROVED** [review]

**Pull request merged**